### ISSUE TITLE
Can't import the named export 'JsonViewer' (imported as 'JsonViewer') from default-exporting module (only default export is available)

### ISSUE THREAD

**Issue by pouu69** [TexteaInc/json-viewer#442]
### issue

- Can't import the named export 'JsonViewer' (imported as 'JsonViewer') from default-exporting module (only default export is available)

### env

- typescript
- react 17
- webpack 5

### sample code

``` js
import { JsonViewer } from '@textea/json-viewer'

export function Test({ data }: { data: object }) {
  return <JsonViewer value={data} />
}

```

Why is the error occurring?

**Comment by pionxzh** [comment]
Can you share the repo or a minimal reproducible stackbliz/codesandbox?

**Comment by pionxzh** [comment]
Try add `"allowSyntheticDefaultImports": true` to your `tsconfig.json`

**Comment by pouu69** [comment]
I can't disclose it because it's a company project, so I'm applying it to nextJS tutorial and sharing it.
- `https://github.com/pouu69/nextjs-dashboard/blob/main/app/page.tsx#L14`
If you check the above repo, the error message is different, but it seems that a similar error occurs in the end.
<img width="2362" alt="image" src="https://github.com/TexteaInc/json-viewer/assets/8802261/d66d1b0f-d58e-4c2f-82cb-dffdbd86af2b">

**Comment by pouu69** [comment]
> Try add `"allowSyntheticDefaultImports": true` to your `tsconfig.json`

The error occurs as usual.

**Comment by pionxzh** [comment]
With the new **App Directory**, React Components are categorized into two: **Server Components** and **Client Components**, and by default they are all Server Component as mentioned in: [Using Server Components in Next.js](https://nextjs.org/docs/app/building-your-application/rendering/server-components#using-server-components-in-nextjs).

Components that need hooks (context, use...) need to be classified as Client Components and they need to have the directive `"use client"` at the top of the source file before any import.

<img width="627" alt="image" src="https://github.com/TexteaInc/json-viewer/assets/9910706/6d6190fd-40a3-489d-aba1-98725bf86506">

**Comment by pionxzh** [comment]
I will add this part to the document and see if this `"use client"` directive should be added by JsonViewer itself.

**Comment by pouu69** [comment]
However, the actual project is not nextjs, but the client component method.

**Comment by pionxzh** [comment]
Can you build a minimum reproduction with the same stack (ts, webpack...etc)?

**Comment by pouu69** [comment]
> Can you build a minimum reproduction with the same stack (ts, webpack...etc)?

I made it into a small structure in the next git.
Another error is occurring this time.

- https://github.com/pouu69/react-json-test

<img width="1893" alt="image" src="https://github.com/TexteaInc/json-viewer/assets/8802261/c5fa64e3-cbab-4f7c-851b-1752b0d04506">

**Comment by pouu69** [comment]
> > Can you build a minimum reproduction with the same stack (ts, webpack...etc)?
> 
> @pionxzh
> 
> I made it into a small structure in the below git. Another error is occurring this time.
> 
> * https://github.com/pouu69/react-json-test
>   
>   * https://github.com/pouu69/react-json-test/blob/main/src/App.tsx#L7
> 
> <img alt="image" width="1893" src="https://private-user-images.githubusercontent.com/8802261/299200987-c5fa64e3-cbab-4f7c-851b-1752b0d04506.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwODU3MjgsIm5iZiI6MTcwNjA4NTQyOCwicGF0aCI6Ii84ODAyMjYxLzI5OTIwMDk4Ny1jNWZhNjRlMy1jYmFiLTRmN2MtODUxYi0xNzUyYjBkMDQ1MDYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjRUMDgzNzA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWQxZWJjNGMxN2Q1YjhmODMyMmYyYjIzN2Y4ZTY0MzZkZWZjNDFhM2Y0ZDJjN2M3ZWVmOTIyZTk2OWNlMzY1NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.8NeUArZUyDE_47qRzXX2Bg6Bdf08brYkUb53qyvLLJE">

I have a fixed above error (screenshot)
I wrote a below cdoe on webpack config

``` js
resolve: {
  alias: {
      'react/jsx-dev-runtime': require.resolve('react/jsx-dev-runtime'),
      'react/jsx-runtime': require.resolve('react/jsx-runtime')
  }
}
```

**Comment by pionxzh** [comment]
What else can I fix for  you? It works ok on my machine.

I also checked the `react/jsx-runtime` error. Seems it can also be fixed by upgrading react to 18. I'm not too sure about the reason.

**Comment by pionxzh** [comment]
After checking the error message, I feel your webpack is choosing the `browser` field in package.json. Will remove it in the next version.

**Comment by pionxzh** [comment]
Can you try to update to v3.3.1 and see if it works?

**Comment by pouu69** [comment]
> What else can I fix for you? It works ok on my machine.
> 
> I also checked the `react/jsx-runtime` error. Seems it can also be fixed by upgrading react to 18. I'm not too sure about the reason.

Did you check if it works on v17 because I use react17?

**Comment by pionxzh** [comment]
Yes, it works on the repo you provided. However, I might still need a reproduction to determine the issue.

Your workaround for the `react/jsx-runtime` error is okay. However, I recommend migrating the project to React 18.

The error message you mentioned in the title shall be fixed by `v3.3.1`. Please test and let me know the result in your project 🙏

**Comment by pionxzh** [comment]
Any update? I will close first due to the inactivity.
